app/src/main/java/com/codingblocks/cbonlineapp/activities/CourseActivity.kt

```diff
@@ -204,85 +212,100 @@ class CourseActivity : AppCompatActivity(), AnkoLogger {
-                    if (course.runs != null)
-                        Clients.api.enrollTrial(course.runs!![0].id!!).enqueue(retrofitCallback { throwable, response ->
-                            if (response?.isSuccessful!!) {
-                                Components.showconfirmation(this, "trial")
-                            }
-                        })
-                }
-                batchAdapter.setData(course.runs!!)
-                buyBtn.setOnClickListener {
-                    if (course.runs != null) {
-                        focusOnView(scrollView, batchRv)
-                    } else {
-                        toast("No available runs right now ! Please check back later")
+                    course.runs?.let { batchAdapter.setData(it) }
+                    buyBtn.setOnClickListener {
+                        if (course.runs != null) {
+                            focusOnView(scrollView, batchRv)
+                        } else {
+                            toast("No available runs right now ! Please check back later")
+                        }
                     }
-                }
-                fetchTags(course)
-                showPromoVideo(course.promoVideo)
-                fetchRating(course.id)
-                if (!course.runs.isNullOrEmpty()) {
-                    val sections = course.runs?.get(0)?.sections
-                    val sectionsList = ArrayList<Sections>()
-                    rvExpendableView.layoutManager = LinearLayoutManager(this)
-                    rvExpendableView.adapter = sectionAdapter
-                    rvExpandableskeleton = Skeleton.bind(rvExpendableView)
-                        .adapter(sectionAdapter)
-                        .shimmer(true)
-                        .angle(20)
-                        .frozen(true)
-                        .duration(1200)
-                        .count(4)
-                        .load(R.layout.item_skeleton_section_card)
-                        .show()
-                    runOnUiThread {
-                        sections!!.forEachIndexed { index, section ->
-                            GlobalScope.launch(Dispatchers.Main) {
-                                val request = service.getSections(section.id!!)
-                                val response = request.await()
-                                if (response.isSuccessful) {
-                                    val value = response.body()!!
-                                    value.order = index
-                                    sectionsList.add(value)
-                                    if (sectionsList.size == sections.size) {
-                                        sectionsList.sortBy { it.order }
-                                        rvExpandableskeleton.hide()
-                                        sectionAdapter.setData(sectionsList)
+                    fetchTags(course)
+                    showPromoVideo(course.promoVideo)
```

Comment:
Even this should be directly fetched from livedata value see MyCourseActitvity

Reply:
let me push an update to this as well